nbs/core.ipynb

```diff
@@ -3204,6 +3321,147 @@
+    "preds = nf.predict(futr_df=AirPassengersPanel_test, conformal_level=[10, 50, 90])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d25b2cd2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#| hide\n",
+    "# test conformal prediction are not applied for models with quantiled-related loss\n",
+    "# by default (ConformalIntervals.enable_quantiles=False)\n",
+    "\n",
+    "conformal_intervals = ConformalIntervals()\n",
+    "\n",
+    "models = []\n",
+    "for nf_model in [NHITS, RNN]:\n",
+    "    params = {\"h\": 12, \"input_size\": 24, \"max_steps\": 1}\n",
+    "    if nf_model.__name__ == \"NHITS\":\n",
+    "        params.update({\"loss\": MQLoss(level=[80])})\n",
+    "    models.append(nf_model(**params))\n",
+    "\n",
+    "\n",
+    "nf = NeuralForecast(models=models, freq='M')\n",
+    "nf.fit(AirPassengersPanel_train, conformal_intervals=conformal_intervals)\n",
+    "preds = nf.predict(futr_df=AirPassengersPanel_test, conformal_level=[10, 50, 90])\n",
+    "\n",
+    "pred_cols = [\n",
+    "    'NHITS-median', 'NHITS-lo-80', 'NHITS-hi-80', 'RNN',\n",
+    "    'RNN-conformal-lo-90', 'RNN-conformal-lo-50', 'RNN-conformal-lo-10',\n",
+    "    'RNN-conformal-hi-10', 'RNN-conformal-hi-50', 'RNN-conformal-hi-90'\n",
+    "]\n",
+    "assert all([col in preds.columns for col in pred_cols])\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7b980087",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#| hide\n",
+    "# test conformal predictions applied to quantiles if ConformalIntervals.enable_quantiles=True\n",
+    "\n",
+    "conformal_intervals = ConformalIntervals(enable_quantiles=True)\n",
+    "\n",
+    "nf = NeuralForecast(models=[NHITS(h=12, input_size=24, max_steps=1, loss=MQLoss(level=[80]))], freq='M')\n",
+    "nf.fit(AirPassengersPanel_train, conformal_intervals=conformal_intervals)\n",
+    "preds = nf.predict(futr_df=AirPassengersPanel_test, conformal_level=[10, 50, 90])\n",
+    "\n",
+    "pred_cols = [\n",
+    "       'NHITS-median', 'NHITS-lo-80', 'NHITS-hi-80',\n",
+    "       'NHITS-median-conformal-lo-90', 'NHITS-median-conformal-lo-50',\n",
+    "       'NHITS-median-conformal-lo-10', 'NHITS-median-conformal-hi-10',\n",
+    "       'NHITS-median-conformal-hi-50', 'NHITS-median-conformal-hi-90',\n",
+    "       'NHITS-lo-80-conformal-lo-90', 'NHITS-lo-80-conformal-lo-50',\n",
+    "       'NHITS-lo-80-conformal-lo-10', 'NHITS-lo-80-conformal-hi-10',\n",
+    "       'NHITS-lo-80-conformal-hi-50', 'NHITS-lo-80-conformal-hi-90',\n",
+    "       'NHITS-hi-80-conformal-lo-90', 'NHITS-hi-80-conformal-lo-50',\n",
+    "       'NHITS-hi-80-conformal-lo-10', 'NHITS-hi-80-conformal-hi-10',\n",
+    "       'NHITS-hi-80-conformal-hi-50', 'NHITS-hi-80-conformal-hi-90'\n",
+    "]\n",
+    "\n",
+    "assert all([col in preds.columns for col in pred_cols])\n"
+   ]
```

Comment:
How do you deal with the case when non-point losses are used in combination with conformal intervals?

Reply:
I think we need more tests, a.o. what exceptions/warnings are raised when we provide combinations of loss functions / prediction intervals that don't work together. But I believe we already capture most of these.

Reply:
I'll resolve this conversation and others to continue [from here](https://github.com/Nixtla/neuralforecast/pull/1171#issuecomment-2399549031)